src/auxiliary.jl

```diff
@@ -71,10 +73,7 @@ Return the coefficient of order `n::Int` of a `a::Taylor1` polynomial.
 get_coeff(a::Taylor1, n::Int) = (@assert 0 ≤ n ≤ a.order+1;
     return a[n+1])
 
-function getindex(a::Taylor1, n::Int)
-    (1 ≤ n ≤ length(a.coeffs)) && return a.coeffs[n]
-    return zero(a.coeffs[1])
-end
+getindex(a::Taylor1, n::Int) = a.coeffs[n]
```

Comment:
I don't think we require checking the bounds because, if we are out of the bounds, an error will be thrown.